src/components/Modals/SlideModal/SlideModal.js

```diff
@@ -327,21 +327,11 @@ class SlideModal extends React.Component<Props, State> {
   }
 }
 
-// withTheme:
-// - is implicitly typed as any;
-// - uses innerRef prop instead of forwarding until v4;
-//   see: https://styled-components.com/docs/api#deprecated-innerref-prop;
-//
-// so the type of wrapped component is set manually to make prop validation
-// of SlideModal work.
-
-type Ref<T> = ((null | T) => mixed) | { current: null | T };
-type ThemedProps = {| ...OwnProps, innerRef: Ref<SlideModal> |};
-const ThemedSlideModal: AbstractComponent<ThemedProps, SlideModal> = withTheme(SlideModal);
+const ThemedSlideModal = withTheme(SlideModal);
 
 // Exported because some calls to React.createRef need this to be passed as type parameter.
 export type SlideModalInstance = SlideModal;
 
 export default React.forwardRef<OwnProps, SlideModalInstance>((props, ref) => (
-  <ThemedSlideModal {...props} innerRef={ref} />
+  <ThemedSlideModal {...props} ref={ref} />
```

Comment:
It's good to see that most of this can finally be removed 😄 However, at the time of writing, there were two separate concerns:

1. `withTheme` wasn't typed, so any component it wrapped lost prop type validation. This is the reason for `AbstractComponent<...>` and still true after the update, which means the default export should still be annotated, only with `AbstractComponent<OwnProps, SlideModal>`.
2. The use of `innerRef` instead of `ref` by styled components. Since this is fixed in the new version, there's no reason to use `React.forwardRef` on our side, and the wrapper component can be removed completely.

To summarise: export `withTheme(SlideModal)` directly, but keep the type annotation.